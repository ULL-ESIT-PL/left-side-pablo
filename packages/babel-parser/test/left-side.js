import { parse } from "../lib";

function getParser(code) {
  return () => parse(code, { sourceType: "module" });
}

describe("left side function assignment", function () {
  it("should parse", function () {
    expect(getParser("foo(5) = 10")()).toMatchSnapshot();
  });
  /*
  it("should not parse", function() {
    expect(getParser("foo.bar(5) = 10")()).toMatchSnapshot();
  });
  */
});
