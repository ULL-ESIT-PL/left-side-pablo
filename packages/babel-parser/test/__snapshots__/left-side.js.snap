// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`left side function assignment should parse 1`] = `
Node {
  "comments": Array [],
  "end": 11,
  "errors": Array [],
  "loc": SourceLocation {
    "end": Position {
      "column": 11,
      "line": 1,
    },
    "start": Position {
      "column": 0,
      "line": 1,
    },
  },
  "program": Node {
    "body": Array [
      Node {
        "end": 11,
        "expression": Node {
          "end": 11,
          "left": Node {
            "arguments": Array [
              Node {
                "end": 5,
                "extra": Object {
                  "raw": "5",
                  "rawValue": 5,
                },
                "loc": SourceLocation {
                  "end": Position {
                    "column": 5,
                    "line": 1,
                  },
                  "start": Position {
                    "column": 4,
                    "line": 1,
                  },
                },
                "start": 4,
                "type": "NumericLiteral",
                "value": 5,
              },
            ],
            "callee": Node {
              "end": 3,
              "loc": SourceLocation {
                "end": Position {
                  "column": 3,
                  "line": 1,
                },
                "identifierName": "foo",
                "start": Position {
                  "column": 0,
                  "line": 1,
                },
              },
              "name": "foo",
              "start": 0,
              "type": "Identifier",
            },
            "end": 6,
            "loc": SourceLocation {
              "end": Position {
                "column": 6,
                "line": 1,
              },
              "start": Position {
                "column": 0,
                "line": 1,
              },
            },
            "start": 0,
            "type": "CallExpression",
          },
          "loc": SourceLocation {
            "end": Position {
              "column": 11,
              "line": 1,
            },
            "start": Position {
              "column": 0,
              "line": 1,
            },
          },
          "operator": "=",
          "right": Node {
            "end": 11,
            "extra": Object {
              "raw": "10",
              "rawValue": 10,
            },
            "loc": SourceLocation {
              "end": Position {
                "column": 11,
                "line": 1,
              },
              "start": Position {
                "column": 9,
                "line": 1,
              },
            },
            "start": 9,
            "type": "NumericLiteral",
            "value": 10,
          },
          "start": 0,
          "type": "AssignmentExpression",
        },
        "loc": SourceLocation {
          "end": Position {
            "column": 11,
            "line": 1,
          },
          "start": Position {
            "column": 0,
            "line": 1,
          },
        },
        "start": 0,
        "type": "ExpressionStatement",
      },
    ],
    "directives": Array [],
    "end": 11,
    "interpreter": null,
    "loc": SourceLocation {
      "end": Position {
        "column": 11,
        "line": 1,
      },
      "start": Position {
        "column": 0,
        "line": 1,
      },
    },
    "sourceType": "module",
    "start": 0,
    "type": "Program",
  },
  "start": 0,
  "type": "File",
}
`;
